### PR TITLE
Set `cancelBuilds` to 0 on close instead of open

### PR DESCRIPTION
**Pull request by ahoppen**
Addresses https://github.com/apple/sourcekit-lsp/pull/1339#discussion_r1612353958

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SourceKitLSP/Swift/SwiftLanguageService.swift [hunk 1]
```diff
@@ -430,6 +429,7 @@ extension SwiftLanguageService {
     let req = sourcekitd.dictionary([
       keys.request: self.requests.editorClose,
       keys.name: uri.pseudoPath,
+      keys.cancelBuilds: 0,
```
Comment (hamishknight) [inline Sources/SourceKitLSP/Swift/SwiftLanguageService.swift:432]: Wasn't the intention to do it for `reopenDocument`? (or should we do it for both?)

Comment (bnbarham) [reply Sources/SourceKitLSP/Swift/SwiftLanguageService.swift:432]: Yeah, I'd say we really only want for re-open

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**